stm/stm.ml

```diff
@@ -496,21 +494,14 @@ end = struct (* {{{ *)
 
   let checkout_shallowest_proof_branch () =
     if List.mem edit_branch (Vcs_.branches !vcs) then begin
-      checkout edit_branch;
-      match get_branch edit_branch with
-      | { kind = `Edit (mode, _,_,_,_) } -> Proof_global.activate_proof_mode mode
-      | _ -> assert false
+      checkout edit_branch
     end else
       let pl = proof_nesting () in
       try
-        let branch, mode = match Vcs_aux.find_proof_at_depth !vcs pl with
-          | h, { Vcs_.kind = `Proof (m, _) } -> h, m | _ -> assert false in
-        checkout branch;
-        stm_prerr_endline (fun () -> "mode:" ^ mode);
-        Proof_global.activate_proof_mode mode
+        let branch = fst @@ Vcs_aux.find_proof_at_depth !vcs pl in
```

Comment:
We have it but I can't stand this syntax. Can you use instead:

    let branch = Vcs_aux.find_proof_at_depth !vcs pl |> fst in

? This is the style that Matej has been introducing bit by bit and I support him on that.